partiql-eval/src/main/kotlin/org/partiql/eval/PartiQLEngine.kt

```diff
@@ -24,7 +24,7 @@ import org.partiql.spi.catalog.Session
  */
 public interface PartiQLEngine {
 
-    public fun prepare(plan: Plan, mode: Mode, session: Session): PartiQLStatement
+    public fun prepare(plan: Plan, session: Session, config: CompilerConfig): PartiQLStatement
```

Comment:
I suggest changing the "Config" to something more like [Context](https://calcite.apache.org/javadocAggregate/org/apache/calcite/plan/Context.html) which may or may not be part of the session, but perhaps these are slightly different?

The Config reads like what should be part of a PartiQLEngine/PartiQLParser/PartiQLPlanner builders. Why not have `PartiQLParser.builder().addListener(...)` because the "config" is what the constructor arguments are for.

 I am curious the use-case of having a changing config for each invocation, and where a session context wouldn't achieve this.

Reply:
Having the error listener as part of a context/config allows for end-users to create singleton planners (if they want to). They're given the freedom to always pass the same object, or to pass in new error listeners.

That being said, I'm up for adding it to the session context -- but I wasn't entirely sure the _right_ place to put it.